Reserved words and expanded FormError use

rushsteve1 opened the pull request:
**Related Issues**
Never opened an issue for this explicitly but it is related to #5 

**Describe the Changes**
Added a new type of `FormError`: `ReservedWord` and a helper function for it. It checks against a list of words names that are not allowed for projects/users/groups/etc. and raises a `FormError` as needed.

With that I have expanded the use of `FormError` to as many forms as I can so that the `ReservedWord` error and other errors can be presented to the user.

**Still To Do**
Expand error messages to other pages more fully.

rushsteve1 committed:
Reserved words and expanded FormError use

Added a new type of FormError: ReservedWord
and a helper function for it. It checks against a list of words names
that are not allowed for projects/users/groups/etc. and raises a
FormError as needed.

With that I have expanded the use of FormError to as many forms as I can
so that the ReservedWord error and other errors can be presented to the user.

BrianMejia commented on src/templates.rs:
```diff
@@ -181,3 +185,13 @@ impl<'v> FromFormValue<'v> for FormError {
+
+pub const RESERVED_WORDS: &[&str] = &["new", "start", "rcos", "edit"];
+
+pub fn is_reserved(word: &str) -> Result<&str, FormError> {
```
In this function, add a check to see if the word is just a number? Having a number could cause unintended effects.

rushsteve1 replied:
Did this in https://github.com/rcos/observatory-new/pull/98/commits/232a1de540eafd3e49d2fe97a679aa7f453fcbab it was only a one-line change.

BrianMejia reviewed (changes requested):
Everything looks to still function, this is a nice catch for stopping unintended redirects. One more case to look out for is to check if the new group/user/event/etc. is just a number.

rushsteve1 committed:
Reserve numbers as well

Names containing only a number could cause conflicts with ID URLs so are
also marked as reserved.

BrianMejia reviewed (approved):
Great! Hopefully no more edge cases show up, but in case it does then it shouldn't be too difficult to solve.

BrianMejia committed:
Merge branch 'master' into reserved-words-formerror
